src/Compilers/VisualBasic/Portable/Parser/ParserFeature.vb

```diff
@@ -153,9 +153,84 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax
+        ''' <typeparam name="TNode"></typeparam>
+        ''' <param name="feature">Language feature to report as unavailable.</param>
+        ''' <param name="options">The parse options being used.</param>
+        ''' <param name="node">The node to attach the diagnostic (Feature Unavailable).</param>
+        ''' <returns>Return the node with the diagnostic attached to it.</returns>
+        <Extension>
+        Friend Function ReportFeatureUnavailable(Of TNode As VisualBasicSyntaxNode)(feature As Feature, options As VisualBasicParseOptions, node As TNode) As TNode
+            Dim featureName = ErrorFactory.ErrorInfo(feature.GetResourceId())
+            Dim requiredVersion As New VisualBasicRequiredLanguageVersion(feature.GetLanguageVersion())
+            Return Parser.ReportSyntaxError(node, ERRID.ERR_LanguageVersion, options.LanguageVersion.GetErrorName(), featureName, requiredVersion)
+        End Function
+
+        ''' <summary>
+        ''' Check to see if the given <paramref name="feature"/> is available with the <see cref="LanguageVersion"/>
+        ''' of the parser.
+        ''' </summary>
+        ''' <param name="feature">Language feature to check is available.</param>
+        ''' <param name="options">The parse options being used.</param>
+        <Extension>
+        Friend Function IsAvailable(feature As Feature, options As VisualBasicParseOptions) As Boolean
+            Return feature.CheckFeatures(options) OrElse feature.IsInLanguageVersion(options)
+        End Function
+
+        ''' <summary>
+        '''  Check to see if the given <paramref name="feature"/> is available with the <see cref="LanguageVersion"/>
+        ''' of the parser.  If it is not available a diagnostic will be added to the returned value.
+        ''' </summary>
+        ''' <param name="node">The node to attach the potential diagnostic (Feature Unavailable).</param>
+        ''' <param name="feature">Language feature to check is available.</param>
+        ''' <param name="options">The parse options being used.</param>
+        <Extension>
+        Friend Function CheckFeatureAvailable(Of TNode As VisualBasicSyntaxNode)(node As TNode, feature As Feature, options As VisualBasicParseOptions) As TNode
+            Return If(feature.IsAvailable(options), node, feature.ReportFeatureUnavailable(options, node))
+        End Function
+
+        ''' <summary>Returns false and reports an error if the feature is unavailable.</summary>
+        ''' <param name="feature">Language feature to check is available.</param>
+        ''' <param name="options">The parse options being used.</param>
+        ''' <param name="diagnostics">The diagnostics to which add this diagnostic.</param>
+        ''' <param name="location">The location to report the diagnostic.</param>
+        <Extension>
+        Friend Function IsAvailable(feature As Feature, options As VisualBasicParseOptions, diagnostics As DiagnosticBag, location As Location) As Boolean
```

Comment:
The method name `IsAvailable` is more appropriate for a method with no side-effects (this method adds to `diagnostics`). For that reason, and for consistency with previous implementations, please rename to `CheckFeatureAvailability`.

Reply:
I agree. This `IsAvailable` should be called `CheckFeatureAvailability` as it used to (since it records diagnostics).

---
In reply to: [132713383](https://github.com/dotnet/roslyn/pull/19319#discussion_r132713383) [](ancestors = 132713383)